foundry/applications/formApplications/baseEntitySheets/journalSheet.d.ts

```diff
@@ -1,23 +1,67 @@
 /**
  * The JournalEntry Configuration Sheet
- *
- * @param entity - The JournalEntry instance which is being edited
- * @param options - Application options
  */
-declare class JournalSheet extends BaseEntitySheet {
+declare class JournalSheet<
+  D extends object = BaseEntitySheet.Data<JournalEntry>,
+  O extends JournalEntry = D extends BaseEntitySheet.Data<infer T> ? T : JournalEntry
+> extends BaseEntitySheet<D, O> {
+  constructor(object: O, options?: JournalSheet.Options);
+
+  /** @override */
+  static get defaultOptions(): JournalSheet.Options;
+
+  /** @override */
+  get id(): string;
+
+  /** @override */
+  get template(): string;
+
+  /** @override */
+  get title(): string;
+
+  /** @override */
+  getData(options?: Application.RenderOptions): Promise<D> | D;
+
   /**
-   * Provide a unique CSS ID for Entity Sheets
+   * Guess the default view mode for the sheet based on the player's permissions to the Entry
    */
-  get id(): string;
+  protected _inferDefaultMode(): JournalSheet.SheetMode;
```

Comment:
This can also return `null`. I don't think you should add `null` to `JournalSheet.SheetMode` because in the calls to `_onSwapMode`, it is never `null`. I'd simply make the return type `JournalSheet.SheetMode | null` here.